### PR TITLE
[ContactStructuralMechanicsApplication] More debug info. Related with the conditions creation and its normals

### PR DESCRIPTION
**Pull request by loumalouomega**
In echo level greater than 0, it creates texts files containing information about the conditions created and its normal

### CODE REVIEW

File: applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp [hunk 1]
```diff
@@ -1492,6 +1493,23 @@ inline void TreeContactSearch<TDim, TNumNodes, TNumNodesMaster>::CreateAuxiliarC
     // Iterate in the conditions and create the new ones
     ConditionsArrayType& r_conditions_array = rContactModelPart.Conditions();
 
+    // In case of debug mode
+    if (mThisParameters["echo_level"].GetInt() > 0) {
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp]: why don't you use `debug_mode` here too?
Then you wouldn't have to manipulate the echo lvl

Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp]: Done

**Comment by loumalouomega** [comment]
Ping @philbucher

File: applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp [hunk 2]
```diff
@@ -1504,6 +1522,22 @@ inline void TreeContactSearch<TDim, TNumNodes, TNumNodesMaster>::CreateAuxiliarC
             }
         }
     }
+
+    // In case of debug mode
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp:1526]: good comment xD

File: applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp [hunk 3]
```diff
@@ -1504,6 +1522,22 @@ inline void TreeContactSearch<TDim, TNumNodes, TNumNodesMaster>::CreateAuxiliarC
...
+    if (mThisParameters["debug_mode"].GetBool()) {
+        std::filebuf debug_buffer;
+        debug_buffer.open("created_conditions_normal_debug.out",std::ios::out);
+        std::ostream os(&debug_buffer);
+        for (auto& cond : rComputingModelPart.Conditions()) {
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp]: const

File: applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp [hunk 4]
```diff
@@ -1492,6 +1493,23 @@ inline void TreeContactSearch<TDim, TNumNodes, TNumNodesMaster>::CreateAuxiliarC
     // Iterate in the conditions and create the new ones
     ConditionsArrayType& r_conditions_array = rContactModelPart.Conditions();
 
+    // In case of debug mode
+    if (mThisParameters["debug_mode"].GetBool()) {
+        std::filebuf debug_buffer;
+        debug_buffer.open("original_conditions_normal_debug.out",std::ios::out);
+        std::ostream os(&debug_buffer);
+        for (auto& cond : r_conditions_array) {
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp]: const

File: applications/ContactStructuralMechanicsApplication/python_scripts/alm_contact_process.py [hunk 5]
```diff
@@ -179,6 +179,36 @@ def ExecuteFinalizeSolutionStep(self):
...
+            total_load[2] = 0.0
+            total_reaction = KM.Vector(3)
+            total_reaction[0] = 0.0
+            total_reaction[1] = 0.0
+            total_reaction[2] = 0.0
+            total_contact_force = 0
+
+            # Computing total load applied (I will consider only surface loads for now)
+            for cond in self.computing_model_part.Conditions:
+                geom = cond.GetGeometry()
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/alm_contact_process.py:196]: is useful right :D

File: applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp [hunk 6]
```diff
@@ -1492,6 +1493,23 @@ inline void TreeContactSearch<TDim, TNumNodes, TNumNodesMaster>::CreateAuxiliarC
     // Iterate in the conditions and create the new ones
     ConditionsArrayType& r_conditions_array = rContactModelPart.Conditions();
 
+    // In case of debug mode
+    if (mThisParameters["debug_mode"].GetBool()) {
+        std::filebuf debug_buffer;
+        debug_buffer.open("original_conditions_normal_debug.out",std::ios::out);
+        std::ostream os(&debug_buffer);
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/custom_utilities/tree_contact_search.cpp:1500]: is this the proper way of opening an output-file in C?
Just curious, good to know if I need this at some point
@roigcarlo ?

**Comment by loumalouomega** [comment]
@philbucher

**Review by philbucher — APPROVED** [review]

**Comment by loumalouomega** [comment]
Thanks

**Pull request merged**